src/app/pipelines/components/pipelines-editor/pipelines-editor.component.ts

```diff
@@ -888,6 +892,15 @@ export class PipelinesEditorComponent implements OnInit {
       (key) => this.designerModel.nodes[key].data.name === node.data.name
     );
     const step = node.data.data;
+    step.params.forEach((parameter) => {
+      if (
+        (parameter.value === '' || !parameter.value) &&
+        parameter.defaultValue &&
+        parameter.required === true
+      ) {
+        parameter.value = parameter.defaultValue;
```

Comment:
There is no need to assign the defaultValue to the value. The back end library chooses which value to use. The purpose of the validation should simply be to decide if the user needs to fill in value. The validation should be:

 if value is empty and defaultValue is empty and required is true, then show a validation error.

The check should be:

if (!parameter.defaultValue && (!parameter.value || (type0f parameter.value === 'string' && parameter.value.trim().length === 0)) {

Reply:
Got it David! I will fix the condition for the error message.